### ISSUE TITLE
Investigate connection reset in DNS helper 

### ISSUE THREAD

**Issue by mderka** [GoogleCloudPlatform/gcloud-java#930]
Without catching an IOException in the DNS helper `handleBatch` method, the `testCreateChangeBatch` causes connection reset.

**Comment by mderka** [comment]
Hint: The problem can be mitigated by simply appending `\r\n` to the response body. For example, change line 262 in `LocalDnsHelper` to:

```java
    Response response(String message) {
      try {
        return new Response(code, toJson(message) + "\r\n");
      } catch (IOException e) {
        return Error.INTERNAL_ERROR.response("Error when generating JSON error response.");
      }
    }
```

Catching the exception is then not necessary. I did not make this change though. If this turns out to be needed, then the `ResourceManagerHelper` should be checked too as I think that DNS uses the same writer implementation. Perhaps these writers could be moved to core.

**Comment by mziccard** [comment]
Fixed with #938
